flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java

```diff
@@ -85,54 +88,75 @@ public void checkForProhibitedLogContents() {
 	public void testDetachedMode() throws InterruptedException, IOException {
 		LOG.info("Starting testDetachedMode()");
 		addTestAppender(FlinkYarnSessionCli.class, Level.INFO);
-		Runner runner =
-			startWithArgs(new String[]{"-j", flinkUberjar.getAbsolutePath(),
-						"-t", flinkLibFolder.getAbsolutePath(),
-						"-n", "1",
-						"-jm", "768",
-						"-tm", "1024",
-						"--name", "MyCustomName", // test setting a custom name
-						"--detached"},
-				"Flink JobManager is now running on", RunTypes.YARN_SESSION);
-
+		File exampleJarLocation = getTestJarPath("StreamingWordCount.jar");
+		// get temporary file for reading input data for wordcount example
+		File tmpInFile;
+		try {
+			tmpInFile = tmp.newFile();
+			FileUtils.writeStringToFile(tmpInFile, WordCountData.TEXT);
+		}
+		catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+		Runner runner = isNewMode ?
```

Comment:
Why do we need this change? Was it always starting in legacy mode before?

Reply:
The reason why the other method always starts a legacy cluster is this line: https://github.com/apache/flink/blob/af5279e9bd3dec18512a54c59982e3dc1f253cd2/flink-yarn-tests/src/test/java/org/apache/flink/yarn/YarnTestBase.java#L531

We always set mode to legacy if there's Kerberos stuff in the config. I don't know yet why that line is there.